File: src/main/java/com/parrit/transformers/ProjectTransformer.java
package com.parrit.transformers;

import com.parrit.DTOs.PairingBoardDTO;
import com.parrit.DTOs.PersonDTO;
import com.parrit.DTOs.ProjectDTO;
import com.parrit.entities.PairingBoard;
import com.parrit.entities.Person;
import com.parrit.entities.Project;

import java.util.List;

public class ProjectTransformer {

    public static ProjectDTO transform(Project project) {
        List<PersonDTO> personDTOs = PersonTransformer.transform(project.getPeople());
        List<PairingBoardDTO> pairingBoardDTOs = PairingBoardTransformer.transform(project.getPairingBoards());
        return new ProjectDTO(project.getId(), project.getName(), pairingBoardDTOs, personDTOs);
    }

    public static Project merge(Project project, ProjectDTO projectDTO) {
        List<PairingBoard> pairingBoards = PairingBoardTransformer.reverse(projectDTO.getPairingBoards());
        List<Person> reversePeople = PersonTransformer.reverse(projectDTO.getPeople());
        return new Project(projectDTO.getName(), project.getPassword(), pairingBoards, reversePeople, project.getId());
    }
}

File: src/main/kotlin/com/parrit/transformers/ProjectTransformer.kt
package com.parrit.transformers

import com.parrit.DTOs.ProjectDTO
import com.parrit.entities.Project

object ProjectTransformer {

    fun transform(project: Project): ProjectDTO {
        val projectDTO = ProjectDTO()
        projectDTO.id = project.id
        projectDTO.name = project.name
        projectDTO.people = PersonTransformer.transform(project.people)
        projectDTO.pairingBoards = PairingBoardTransformer.transform(project.pairingBoards)
        return projectDTO
    }

    fun merge(project: Project, projectDTO: ProjectDTO): Project {
        val mergedProject = Project()
        mergedProject.id = project.id
        mergedProject.name = projectDTO.name
        mergedProject.password = project.password
        mergedProject.people = PersonTransformer.reverse(projectDTO.people)
        mergedProject.pairingBoards = PairingBoardTransformer.reverse(projectDTO.pairingBoards)
        return mergedProject
    }
}
